src/machines/commerce-machine.ts

```diff
@@ -182,9 +183,27 @@ export const commerceMachine = createMachine<
             return {}
           }
 
+          const result = z
+            .object({plans: z.array(pricingPlanSchema)})
+            .safeParse(context.pricingData)
+          if (!result.success) return {pricingData: event.data}
+
+          // Coupons with a minimum can be applicable on a per-plan basis. If
+          // there is no price_discounted/price_savings for a specific plan
+          // (`priceId`), then we should NOT include the `couponToApply`.
+          const currentPlan = result.data.plans.find(
+            (plan) => plan.stripe_price_id === context.priceId,
+          )
+          const noCouponToApply = isEmpty(currentPlan?.price_discounted)
+
+          const couponData: {} | {couponToApply: undefined | object} =
+            noCouponToApply
+              ? {couponToApply: undefined}
+              : extractAppliedDefaultCoupon(event.data)
```

Comment:
Without this ternary, we are overeagerly applying the default/site-wide coupon to plans for which it doesn't apply. If we detect that the price of the current plan hasn't been discounted, then we know the default coupon is _not_ applied. So we can ensure it isn't set (hence, `{couponToApply: undefined}`).